lib/src/repository.rs

```diff
@@ -27,8 +31,40 @@ pub trait RepositoryExt {
     /// For a given message of an issue, find the initial message.
     ///
     fn find_tree_init<'a>(&'a self, commit: Commit<'a>) -> Result<Commit>;
+
+    /// Get issue hashes for a prefix
+    ///
+    /// This function returns all known issues known to the DIT repo under the
+    /// prefix provided (e.g. all issues for which refs exist under
+    /// `<prefix>/dit/`). Provide "refs" as the prefix to get only local issues.
+    ///
+    fn get_issue_hashes(&self, prefix: &str) -> Result<HashIterator>;
+
+    /// Get all issue hashes
+    ///
+    /// This function returns all known issues known to the DIT repo.
+    ///
+    fn get_all_issue_hashes(&self) -> Result<HashIterator>;
+}
+
+
+/// Get issue hashes from head references
+///
+/// Retrieve the issue hashes from head references provided. References which
+/// are not head references are ignores. However, the function does not check
+/// whether the references are, in fact, references in a dit namespace.
+///
+fn head_refs_to_issues(refs: References) -> HashIterator {
+    let hashes = refs.names()
+                     .filter_map(|name| name.ok()) // TODO: propagate errors
+                     .filter(|name| name.ends_with("/head"))
+                     .filter_map(|name| name.rsplitn(3, "/")
+                                            .nth(1)
+                                            .and_then(|hash| Oid::from_str(hash).ok()));
+    HashIterator::new(hashes)
```

Comment:
If you implemennt `From<I> for BoxedIterator<I> where I: Iterator<...>` we could reduce this boilerplate and implement this function as:

```rust
refs.names()
     .filter_map(|name| name.ok()) // TODO: propagate errors
     .filter(|name| name.ends_with("/head"))
     .filter_map(|name| name.rsplitn(3, "/").nth(1).and_then(|hash| Oid::from_str(hash).ok()));
```

I guess.